go/WorkQueue.go

```diff
@@ -120,6 +122,67 @@ func (workQueue *WorkQueue) AddItem(ctx context.Context, db *redis.Client, item
 	return err
 }
 
+func (workQueue *WorkQueue) AddAtomicItem(ctx context.Context, db *redis.Client, item Item) error {
+	tx := db.TxPipeline()
+
+	// Add LRANGE commands to the transaction
+	tx.LRange(ctx, workQueue.mainQueueKey, 0, -1)
+	tx.LRange(ctx, workQueue.processingKey, 0, -1)
+
+	// Execute the transaction
+	results, err := tx.Exec(ctx)
+	if err != nil {
+		return err
+	}
+
+	mainItems := results[0].(*redis.StringSliceCmd).Val()
+	processingItems := results[1].(*redis.StringSliceCmd).Val()
+	itemID := item.ID
+
+	if sliceContainsString(mainItems, itemID) || sliceContainsString(processingItems, itemID) {
+		// Same item tried being added twice.
+		return nil
+	}
+
+	pipeline := db.TxPipeline()
+	workQueue.AddItemToPipeline(ctx, pipeline, item)
+	_, err = pipeline.Exec(ctx)
+	return err
+}
+
```

Comment:
What happens if, in between the first transaction (on line 133) and adding the item (on line 149), another thread comes and adds the item?

Also, the point of this method is to add an item only if it's not a duplicate. I think the name of the method should make that obvious.